pkg/storage/utils/decomposedfs/decomposedfs.go

```diff
@@ -210,9 +215,31 @@ func (fs *Decomposedfs) CreateHome(ctx context.Context) (err error) {
 			return
 		}
 	}
+
+	// add storage space
+	if err := fs.createStorageSpace("personal", h.ID); err != nil {
```

Comment:
`createStorageSpace` is called at the time of home and share creation and in `Tree.Setup`, you create spaces for existing nodes, but these won't be created when created directories and uploading files.

Reply:
That works as designed. A new directory is not automatically a new storage space. Neither is a file. They don't contain any new sharing information (ACLs) or a quota. Maybe we need to clarify the meaning of a space.

`Tree.Setup` checks if the spaces folder exists and if not will pick up all user homes and shared nodes to create `personal` and `share` type spaces. We do this to migrate an existing layout to now also keep track of spaces.

I consider `CreateHome` a legacy method to provision user homes. IMO the [admin `CreateUser` api](https://cs3org.github.io/cs3apis/#cs3.admin.user.v1beta1.CreateUserRequest) should, as part of creating the user, provision the users home or `personal` user spaces using the yet to implement `CreateStorageSpace` call. But that is part of a subsequent PR. For now we just want to make the different types of entrypoints (user homes, shares) discoverable as spaces.

other types could be `project`, `music` or `archive` ... far future though ... we'll see how it evolves.

Reply:
Ah okay. For some reason, I thought each resource (or node in the ocis fs) is considered a space